src/text_render.rs

```diff
@@ -94,13 +97,16 @@ impl TextRenderer {
         &mut self,
         device: &Device,
         queue: &Queue,
+        encoder: &mut CommandEncoder,
         font_system: &mut FontSystem,
         atlas: &mut TextAtlas,
         screen_resolution: Resolution,
         text_areas: impl IntoIterator<Item = TextArea<'a>>,
         cache: &mut SwashCache,
         mut metadata_to_depth: impl FnMut(usize) -> f32,
     ) -> Result<(), PrepareError> {
+        self.staging_belt.recall();
```

Comment:
Technically, for lowest memory usage `recall` should be called right after the user submits the `Queue`. But this approach removes the burden of a `StagingBelt` from the user and keeps it hidden.

Worst case scenario, we allocate some additional chunks. Overhead should be constant.